klippy/extras/homing_heaters.py

```diff
@@ -0,0 +1,54 @@
+# Heater handling on homing moves
+#
+# Copyright (C) 2016-2018  Kevin O'Connor <kevin@koconnor.net>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+class HomingHeaters:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.printer.register_event_handler("homing:move_begin",
+                                            self.handle_homing_move_begin)
+        self.printer.register_event_handler("homing:move_end",
+                                            self.handle_homing_move_end)
+        self.heaters_to_disable = config.get("heaters", "")
+        self.steppers_needing_quiet = config.get("steppers", "")
```

Comment:
As a minor suggestion, you may want to validate the config at startup.  (Users prefer getting an error at startup then an error (and/or a failure) during runtime.)  To do this, you can use `self.printer.register_event_handler("klippy:connect")` to register a connect callback.  In that callback it is then possible to lookup the actual heater and stepper objects.

Reply:
I'll work on that

Reply:
commit 87516f2cd33e0bf2d89ac7e9f4714c2c3dfefba3